### PR TITLE
Intro to Auth -Joseph Bradley - Braden Walker

### PR DESCRIPTION
**Pull request by jmbradley**

### CODE REVIEW

**Comment by jmbradley** [comment]
Adding in cookies and cookie management.

File: Auth/index.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  cors(),
+  session(sessionConfig),
+);
+
+server.get('/', (req, res) => {
+  res.send('Werk Werk');
+});
+
+server.post('/register', (req, res) => {
+  const credentials = req.body;
```
Comment (Bradiowave) [inline Auth/index.js]: Specify that you only want the username and password from the request body.

File: Auth/index.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+server.post('/register', (req, res) => {
+  const credentials = req.body; 
+
+  const hash = bcrypt.hashSync(credentials.password, 14);
+  credentials.password = hash;
+
+  db('users').insert(credentials).then(ids => {
+    const id = ids[0];
+    req.session.username = credentials.username;
```
Comment (Bradiowave) [inline Auth/index.js]: Good work signing the user in after they have registered.

File: Auth/index.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  const hash = bcrypt.hashSync(credentials.password, 14);
+  credentials.password = hash;
+
+  db('users').insert(credentials).then(ids => {
+    const id = ids[0];
+    req.session.username = credentials.username;  
+    res.status(201).json({ newUserId: id})
+  })
+   .catch(err => {
+     res.status(500).json(err);
```
Comment (Bradiowave) [inline Auth/index.js]: Write your own error message to explain why the user couldn't register.

**Pull request closed without merging**